src/compiler/parser.ts

```diff
@@ -4539,10 +4539,17 @@ namespace ts {
             //  - "(x,y)" is a comma expression parsed as a signature with two parameters.
             //  - "a ? (b): c" will have "(b):" parsed as a signature with a return type annotation.
             //  - "a ? (b): function() {}" will too, since function() is a valid JSDoc function type.
-            //  - "a ? (b): (function() {})" as well, but inside of a parenthesized type.
+            //  - "a ? (b): (function() {})" as well, but inside of a parenthesized type with an arbitrary amount of nesting.
             //
             // So we need just a bit of lookahead to ensure that it can only be a signature.
-            const hasJSDocFunctionType = type && (isJSDocFunctionType(type) || isParenthesizedTypeNode(type) && isJSDocFunctionType(type.type));
+
+            let unwrappedType = type;

```

Comment:
It seems you can use this helper

https://github.com/microsoft/TypeScript/blob/1ade73df2bf3e83085ca4647d0a3339abe3f869b/src/compiler/utilities.ts#L2918-L2920

Reply:
That walks upward, which is the wrong direction. I need to remove the parens, not walk up from the inside to the outermost parens.

Reply:
Got it. Thanks for the clarification.